### PR TITLE
Convert WarningPanel to TypeScript (#554)

### PR DESCRIPTION
**Pull request by Koroeskohr**
## Hey, I just made a Pull Request!

One thing, propTypes pointed to `message` prop being required, but one story was not specifying it. I assumed it was actually not required in the type definition, feel free to correct me

#### :heavy_check_mark: Checklist

- [x] All tests are passing `yarn test`
- [ ] Screenshots attached (for UI changes)
- [ ] Relevant documentation updated
- [x] Prettier run on changed files
- [ ] Tests added for new functionality
- [ ] Regression tests added for bug fixes

### CODE REVIEW

File: packages/core/src/components/WarningPanel/WarningPanel.tsx [hunk 1]
```diff
@@ -47,34 +48,35 @@ const styles = theme => ({
...
-          <ErrorOutlineStyled />
-          <Typography className={classes.headerText} variant="subtitle1">
-            {title}
-          </Typography>
-        </div>
-        {message && (
-          <Typography className={classes.messageText}>{message}</Typography>
-        )}
-        {children}
+type Props = CSS.Properties & {
```
Comment (Rugvip) [inline packages/core/src/components/WarningPanel/WarningPanel.tsx:58]: How come you have `CSS.Properties` here?

**Review by Rugvip — COMMENTED** [review]
Thanks! 👍 

I thing to clear up and I'd say good to merge

File: packages/core/src/components/WarningPanel/WarningPanel.tsx (hunk 1 above)
Comment (Koroeskohr) [reply packages/core/src/components/WarningPanel/WarningPanel.tsx]: Fixed

**Review by stefanalund — APPROVED** [review]
Nice! Thank you. Happy first contribution 🎉

**Pull request merged**